### PR TITLE
Add the PBC hopper results to the hopper example

### PR DESCRIPTION
**Pull request by acdaigneault**
# Description of the problem

- There was no example to use the periodic boundaries in DEM or CFD-DEM

# Description of the solution

- Since PBC was tested with the hopper, it is a good example to expand to show how to use the parameters

# How Has This Been Tested?

- No needs, it's only doc

# Documentation

- 3D rectangular hopper example in DEM

### CODE REVIEW

File: doc/source/examples/dem/rectangular-hopper/rectangular-hopper.rst [hunk 1]
```diff
@@ -4,14 +4,15 @@ Rectangular hopper
 
 This example simulates the filling and discharging of particles in a rectangular hopper.
 We set up this simulation based on the simulation of Anand et al. `[1] <https://doi.org/10.1016/j.ces.2008.08.015>`_. It is recommended to visit `DEM parameters <../../../parameters/dem/dem.html>`_ for more detailed information on the concepts and physical meanings of the parameters in Lethe-DEM.
+Main example do not use periodic boundary conditions in depth of the hopper as the article, but the extensional case does.
```
Comment (blaisb) [inline doc/source/examples/dem/rectangular-hopper/rectangular-hopper.rst:7]:
```suggestion
The main example does not use periodic boundary conditions in the depth of the hopper like article, but an extended case, presented at the end, does.
```

File: doc/source/examples/dem/rectangular-hopper/rectangular-hopper.rst [hunk 2]
```diff
@@ -234,6 +235,117 @@ The simulated mass discharging rate is 84.94 g/s.
...
+~~~~~~~~~~~~~~~~~~
+Here is the comparison of the results from the original simulation with Lethe DEM, the simulation with periodic boundary conditions with Lethe and the results from Anand et al. paper.
+The simulated mass discharging rate is 84.94 g/s from the original simulation and 88.77 g/s with PBC.
+
+.. figure:: images/figure_hopper_comparison.png
+    :width: 600
+    :alt: Comparison
+    :align: center
+
+    Comparison of mass discharge results from the 2 simulations and the paper.
```
Comment (blaisb) [inline doc/source/examples/dem/rectangular-hopper/rectangular-hopper.rst:347]:
```suggestion
    Comparison of mass discharge results from the 2 simulations and the journal article.
```

File: doc/source/examples/dem/rectangular-hopper/rectangular-hopper.rst [hunk 3]
```diff
@@ -234,6 +235,117 @@ The simulated mass discharging rate is 84.94 g/s.
...
+
```
Comment (blaisb) [inline doc/source/examples/dem/rectangular-hopper/rectangular-hopper.rst:348]: Can you maybe add a comparison of the run time? I presume the periodic case has a much lower run time no?

**Review by blaisb — APPROVED** [review]
Minor small changes. Very good extension, great work :)!

File: doc/source/examples/dem/rectangular-hopper/rectangular-hopper.rst (hunk 3 above)
Comment (acdaigneault) [reply doc/source/examples/dem/rectangular-hopper/rectangular-hopper.rst:348]: Yes from about 3h to 30 min

File: doc/source/examples/dem/rectangular-hopper/rectangular-hopper.rst [hunk 4]
```diff
@@ -234,6 +235,117 @@ The simulated mass discharging rate is 84.94 g/s.
...
+        set insertion box maximum z                        = 0.03136
+        set insertion distance threshold                   = 1.5
+        set insertion random number range                  = 0.1
+        set insertion random number seed                   = 20
+    end
+
+Results comparison
+~~~~~~~~~~~~~~~~~~
+Here is the comparison of the results from the original simulation with Lethe DEM, the simulation with periodic boundary conditions with Lethe and the results from Anand et al. paper.
+The simulated mass discharging rate is 84.94 g/s from the original simulation and 88.77 g/s with PBC.
```
Comment (acdaigneault) [inline doc/source/examples/dem/rectangular-hopper/rectangular-hopper.rst:340]:
```suggestion
The simulated mass discharging rate is 84.94 g/s from the original simulation and 88.77 g/s with PBC. Also, the run time of the simulation goes from about 3 hours and 15 minutes to 30 minutes on 8 cores.
```

**Review by blaisb — APPROVED** [review]

**Pull request merged**